### PR TITLE
fix: only send public messages to channel members

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: server/packet_handlers.py [hunk 1]
```diff
@@ -123,16 +123,20 @@ async def send_public_message_handler(session: "Session", packet_data: bytes):
...
     )
 
-    for other_session in await sessions.fetch_all(osu_clients_only=True):
-        if other_session["session_id"] == session["session_id"]:
+    channel = await channels.fetch_one_by_name(channel_name)
+    if channel is None:
+        return
+
+    for session_id in await channel_members.members(channel["channel_id"]):
+        if session_id == session["session_id"]:
```
Comment (cmyui) [inline server/packet_handlers.py]:
```suggestion
    for other_session_id in await channel_members.members(channel["channel_id"]):
        if other_session_id == session["session_id"]:
```

File: server/packet_handlers.py [hunk 2]
```diff
@@ -123,16 +123,20 @@ async def send_public_message_handler(session: "Session", packet_data: bytes):
...
             continue
 
         await packet_bundles.enqueue(
-            other_session["session_id"],
+            session_id,
```
Comment (cmyui) [inline server/packet_handlers.py]:
```suggestion
            other_session_id,
```

**Review by metal-face — APPROVED** [review]

**Comment by cmyui** [comment]
needs testing

**Comment by cmyui** [comment]
redoing this in #41

**Pull request closed without merging**